### PR TITLE
Add miminal internal documentation

### PR DESCRIPTION
**Pull request by moio**
Just a few documents to make it easier to navigate the project structure.

### CODE REVIEW

**Comment by moio** [comment]
Oops, misclicked. @chiaradiamarcelo I will correct anything you note here, sorry!!!

File: DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Design considerations
+
+- this app tries to conform to the [Twelve Factor-App](https://12factor.net/) manifesto
+  - [currently with a known limitation according to principle VI](https://github.com/uyuni-project/hub-xmlrpc-api/issues/56)
+- this app's code is structured according to the [Clean Code Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html)
+- this app fundamentally implements a [Gorilla](https://www.gorillatoolkit.org/) XMLRPC Server to serve its API
```
Comment (chiaradiamarcelo) [inline DESIGN.md:6]: Actually we implement the codec. The server is provided already by Gorilla RPC

File: DESIGN.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+- it uses [kolo/xmlrpc](https://github.com/kolo/xmlrpc) as the XMLRPC client library to consume other XMLRPC APIs
```
Comment (chiaradiamarcelo) [inline DESIGN.md:7]: This library is extended on our side to decode server requests and encode server responses. Not sure if it's worth mentioning it

File: controller/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Controllers are called by the Codec Gorilla component and marshal/unmarshal requests and responses to functions in the gaeway package. There is one Controller per supported method or namespace.
```
Comment (chiaradiamarcelo) [inline controller/README.md:1]: typo: gateway

File: controller/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+
+Functions in the gateway package actually implement that method or namespace's functionality.
```
Comment (chiaradiamarcelo) [inline controller/README.md:3]: I would add: Functions in the gateway package implement the bussines rules, meaning all the use cases

**Comment by chiaradiamarcelo** [comment]
Awesome @moio

File: controller/README.md (hunk 3 above)
Comment (moio) [reply controller/README.md:1]: Fixed

File: controller/README.md (hunk 4 above)
Comment (moio) [reply controller/README.md:3]: Fixed

**Pull request merged**